server/src/internalClusterTest/java/org/opensearch/ingest/IngestClientIT.java

```diff
@@ -209,6 +223,103 @@ public void testBulkWithIngestFailures() throws Exception {
         assertTrue(deletePipelineResponse.isAcknowledged());
     }
 
+    public void testBulkWithIngestFailuresBatch() throws Exception {
```

Comment:
I'm not sure we need this test in addition to `runBulkTestWithRandomDocs`. What do you think? I made it as an example of the simplest case that would fail to illustrate the bug, but `runBulkTestWithRandomDocs` with a batch size > 1 also failed before this fix.

Reply:
Agreed, can I raise a separate PR to remove this test and modify the consumer mock later? Because another commit could trigger build and flaky tests can really delay this fix.